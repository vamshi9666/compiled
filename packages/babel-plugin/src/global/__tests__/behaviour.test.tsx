import { transformSync } from '@babel/core';
import babelPlugin from '../../index';

const transform = (code: TemplateStringsArray) => {
  return transformSync(code[0], {
    configFile: false,
    babelrc: false,
    highlightCode: false,
    plugins: [babelPlugin],
  })?.code;
};

describe('global component transform', () => {
  it('should transform global component', () => {
    const actual = transform`
      import { Global } from '@compiled/react';

      <Global styles={{ fontSize: 20 }} />
    `;

    expect(actual).toMatchInlineSnapshot(`
      "/* File generated by @compiled/babel-plugin v0.0.0 */

      import * as React from 'react';
      import { ax, ix, CC, CS } from \\"@compiled/react/runtime\\";
      const _ = \\":root{font-size:20px}\\";
      <CC>
          <CS>{[_]}</CS>
        </CC>;"
    `);
  });

  it('should replace the global component with null if there is no style prop', () => {
    const actual = transform`
      import { Global } from '@compiled/react';

      <Global />
    `;

    expect(actual).toMatchInlineSnapshot(`
      "/* File generated by @compiled/babel-plugin v0.0.0 */

      import * as React from 'react';
      import { ax, ix, CC, CS } from \\"@compiled/react/runtime\\";
      null;"
    `);
  });

  it('should replace the global component with null if there are no defined styles', () => {
    const actual = transform`
      import { Global } from '@compiled/react';

      <Global styles={{}} />
    `;

    expect(actual).toMatchInlineSnapshot(`
      "/* File generated by @compiled/babel-plugin v0.0.0 */

      import * as React from 'react';
      import { ax, ix, CC, CS } from \\"@compiled/react/runtime\\";
      null;"
    `);
  });

  it('should throw if any dynamic declarations were found', () => {
    expect(() => {
      transform`
        import { Global } from '@compiled/react';

        let fontSize = 10;
        fontSize = 20;

        <Global styles={{ fontSize }} />
      `;
    }).toThrowErrorMatchingSnapshot();
  });

  it('should throw if any dynamic declarations were found', () => {
    expect(() => {
      transform`
        import { Global } from '@compiled/react';

        const isSomething = true;
        <Global styles={[isSomething && { fontSize: 20 }]} />
      `;
    }).toThrowErrorMatchingSnapshot();
  });
});
