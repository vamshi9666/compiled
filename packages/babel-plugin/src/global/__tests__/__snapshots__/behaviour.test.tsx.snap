// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global component transform should throw if any dynamic declarations were found 1`] = `
"unknown: Global component doesn't support dynamic CSS please refactor it to be static (7:26).
  5 |         fontSize = 20;
  6 | 
> 7 |         <Global styles={{ fontSize }} />
    |                           ^^^^^^^^
  8 |       "
`;
